File: Data Structures & Algorithms/missing-number/submission-6.cpp
class Solution {
private:
    void bubbleSort(vector<int>& nums) {
    int n = nums.size();

    for (int i = 0; i < n - 1; i++) {
        bool swapped = false;

        for (int j = 0; j < n - 1 - i; j++) {
            if (nums[j] > nums[j + 1]) {
                swap(nums[j], nums[j + 1]);
                swapped = true;
            }
        }

        // Already sorted
        if (!swapped) {
            break;
        }
    }
}

public:
    int missingNumber(vector<int>& nums) {
        int result = nums.size();

    for (int i = 0; i < nums.size(); i++) {
        result ^= i;
        result ^= nums[i];
    }

    return result;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-6.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        if len(nums) == 1:
            return nums[0]

        while True:
            print("mid")
            if r - 1 == l:
                return min(nums[l], nums[r])
            mid = ((r - l)//2) + l
            print(mid)

            if nums[mid] > nums[mid + 1]:
                return nums[mid+1]
            elif nums[l] > nums[mid]:
                r = mid
            elif nums[r] < nums[mid]:
                l = mid
            else:
                return nums[l]
